### PR TITLE
Update docstrings with information about static build

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-69190

### CODE REVIEW

File: src/inference/include/openvino/runtime/core.hpp [hunk 1]
```diff
@@ -43,14 +43,17 @@ class OPENVINO_RUNTIME_API Core {
...
-     * See Core::register_plugins for more details.
+     * There are 2 ways how Core configures device plugins:
+     * 1. (default) Use XML configuration file in case of dynamic libraries build;
+     * 2. Use strictly defined configuration in case of static libraries build.
      *
      * @param xml_config_file Path to the .xml file with plugins to load from. If the XML configuration file is not
-     * specified, default OpenVINO Runtime plugins are loaded from the default `plugin.xml` file located in the same
-     * folder as OpenVINO runtime shared library.
+     * specified, default OpenVINO Runtime plugins are loaded from:
+     * 1. (dynamic build) default `plugin.xml` file located in the same folder as OpenVINO runtime shared library;
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/core.hpp:55]: plugin.xml => plugins.xml

File: src/inference/include/openvino/runtime/core.hpp [hunk 2]
```diff
@@ -43,14 +43,17 @@ class OPENVINO_RUNTIME_API Core {
     std::shared_ptr<Impl> _impl;
 
 public:
-    /** @brief Constructs an OpenVINO Core instance using the XML configuration file with
-     * devices and their plugins description.
+    /** @brief Constructs an OpenVINO Core instance with devices
+     * and their plugins description.
      *
-     * See Core::register_plugins for more details.
+     * There are 2 ways how Core configures device plugins:
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/core.hpp:49]: Let's use `2` => `two`

File: src/inference/include/ie/ie_core.hpp [hunk 3]
```diff
@@ -34,13 +34,17 @@ class INFERENCE_ENGINE_API_CLASS(Core) {
     std::shared_ptr<Impl> _impl;
 
 public:
-    /** @brief Constructs Inference Engine Core instance using XML configuration file with
-     * plugins description.
+    /** @brief Constructs an Inference Engine Core instance with devices
+     * and their plugins description.
      *
-     * See RegisterPlugins for more details.
+     * There are 2 ways how Core configures device plugins:
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/ie_core.hpp:40]: `configures ` => `to configure `

File: src/inference/include/openvino/runtime/core.hpp (hunk 1 above)
Comment (vurusovs) [reply src/inference/include/openvino/runtime/core.hpp]: Thanks! Fixed

File: src/inference/include/openvino/runtime/core.hpp (hunk 2 above)
Comment (vurusovs) [reply src/inference/include/openvino/runtime/core.hpp]: Thanks! Fixed

File: src/inference/include/ie/ie_core.hpp (hunk 3 above)
Comment (vurusovs) [reply src/inference/include/ie/ie_core.hpp]: Thanks! Fixed

**Pull request merged**